plugins/woocommerce/includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -378,6 +380,8 @@ public function decrease_usage_count( &$coupon, $used_by = '' ) {
 			}
 		}
 
+		$this->refresh_coupon_data( $coupon );
```

Comment:
I'm not sure if I understand the issue fully, but while `decrease_usage_count` does contain raw SQL, that's only a `SELECT`. The actual changes (and therefore the cache invalidation) are happening via `update_usage_count_meta`, `delete_metadata_by_mid`, and `$coupon->set_used_by`. 

I see three lines in which we change data in this function:

- `$this->update_usage_count_meta` (this already calls `refresh_coupon_data` itself)
- `delete_metadata_by_mid` (calls `wp_cache_delete`)
- `$coupon->set_used_by` (calls `set_prop`, which does not actually change data in the database -- not super clear to me when and where this actually gets written to the database?)

Do we really need to `refresh_coupon_data` here? 🤔 Or do we maybe only need it inside of the `if ( $meta_id )` block?